File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 品牌接口
 * @author Administrator
 *
 */
public interface BrandService {
	//查询全部
	public List<TbBrand> findAll();
	public PageResult findPage(int pageNum, int pageSize);
	//增加
	public void add(TbBrand brand);
	//查询
	public TbBrand findOne(Long id);
	//修改
	public void update(TbBrand brand);
	//删除
	public void dele(Long[] ids);
	//条件查询
	public PageResult findPage(TbBrand brand, int pageNum, int pageSize);
	//下拉列表
	public List<Map> selectOptionList();


}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service('brandService',function($http){

    //读取列表数据绑定到表单中
    this.findAll=function(){
        return $http.get('../brand/findAll.do');
    }
    //分页
    this.findPage=function(page,size){
        return $http.get('../brand/findPage.do?page='+page+'&size='+size);
    }
    //查询实体
    this.findOne=function(id){
        return $http.get('../brand/findOne.do?id='+id);
    }
    //增加
    this.add=function(entity){
        return  $http.post('../brand/add.do',entity );
    }
    //修改
    this.update=function(entity){
        return  $http.post('../brand/update.do',entity );
    }
    //删除
    this.dele=function(ids){
        return $http.get('../brand/dele.do?ids='+ids);
    }
    //条件查询
    this.search=function(page,size,searchEntity){
        return $http.post('../brand/search.do?page='+page+"&size="+size, searchEntity);
    }
    //下拉列表的数据
    this.selectOptionList=function () {
        return $http.get('../brand/selectOptionList.do');
    }

});